python/test/test_add_users_all_org_members_github_team.py

```diff
@@ -1,26 +1,59 @@
 import os
 import unittest
-from unittest.mock import patch
-
-from github import Github
-from gql import Client
-from gql.transport.aiohttp import AIOHTTPTransport
+from unittest.mock import MagicMock, patch
 
 from python.scripts import add_users_all_org_members_github_team
 
 
-@patch.object(Github, "__new__")
-@patch.object(AIOHTTPTransport, "__new__")
-@patch.object(Client, "__new__")
-class TestAddUsersEveryoneGithubTeam(unittest.TestCase):
+@patch("github.Github.__new__", new=MagicMock)
+@patch("gql.Client.__new__", new=MagicMock)
+@patch("gql.transport.aiohttp.AIOHTTPTransport.__new__", new=MagicMock)
+class TestAddUsersEveryoneGithubTeamMain(unittest.TestCase):
 
-    @patch.dict(os.environ, {"ADMIN_GITHUB_TOKEN": "token"})
-    def test_main_smoke_test(self, mock1, mock2, mock3):
+    @patch.dict(os.environ, {"ADMIN_GITHUB_TOKEN": "token", "GITHUB_ORGANIZATION_NAME": "ministryofjustice"})
+    def test_main_smoke_test(self):
         add_users_all_org_members_github_team.main()
 
-    def test_main_returns_error_when_no_token_provided(self, mock1, mock2, mock3):
+
+class TestAddUsersEveryoneGithubTeamGetEnvironmentVariables(unittest.TestCase):
+    def test_raises_error_when_no_environment_variables_provided(self):
+        self.assertRaises(
+            ValueError, add_users_all_org_members_github_team.get_environment_variables)
+
+    @patch.dict(os.environ, {"GITHUB_ORGANIZATION_NAME": "ministryofjustice"})
+    def test_raises_error_when_no_github_token(self):
+        self.assertRaises(
+            ValueError, add_users_all_org_members_github_team.get_environment_variables)
```

Comment:
Duplicate tests?

Reply:
One of the linters also thought the same! 😁

Reply:
yeah sonarcloud compains I closed it as wont fix